src/app/query/bns/bns.query.ts

```diff
@@ -14,13 +14,16 @@ const bnsQueryOptions = {
 } as const;
 
 function getBnsNameFetcherFactory(api: Api) {
-  return (address: string) => () =>
-    api.namesApi.getNamesOwnedByAddress({ address, blockchain: 'stacks' });
+  return (address?: string) => () => {
+    if (!address) return;
+    return api.namesApi.getNamesOwnedByAddress({ address, blockchain: 'stacks' });
+  };
 }
 
-export function useGetBnsNamesOwnedByAddress(address: string) {
+export function useGetBnsNamesOwnedByAddress(address?: string) {
```

Comment:
Same here, can we not make address optional? 🤔  

I'm familiar with the trouble, which boils down to hooks not being able to be called conditionally, but the code will be in a better state if we can avoid making every hook support `undefined` input. Maybe we can jam through some ideas how to better approach this app wide.